### DISCUSSION TITLE
How to ignore serverName in tls connection?

### DISCUSSION THREAD

**Discussion by AlexeySa** [XTLS/Xray-core#3537] [Q&A]
I want to use a TLS connection with a self-signed certificate (CN=localhost). I need to use any `serverName` with this certificate.
How can I make Xray ignore the domain and use my certificate which I manually specified?
I only need tls-connection encryption.

Thanks for the help!

My config:
```
"streamSettings": {
	"network": "ws",
	"security": "tls",
	"tlsSettings": {
	  "allowInsecure": false,
	  "disableSystemRoot": true,
	  "serverName": "microsoft.com",
	  "alpn": ["h2", "http/1.1"],
	  "certificates": [{
		"oneTimeLoading": true,
		"usage": "encipherment",
		"certificateFile": "/opt/xray/cert.crt",
		"keyFile": "/opt/xray/cert.key"
	  }]
	}
}
```
Error:
`2024/07/15 05:13:34 [Error] transport/internet/websocket: failed to dial to 123.45.67.89:443 > tls: failed to verify certificate: x509: certificate is not valid for any names, but wanted to match microsoft.com`

**Comment by uuonda** [comment]
I don't think you can unless you are willing to modify sources. But you probably can issue a certificate on the fly with your own CA.

**Comment by kingwilliam** [comment]
on client side

`"allowInsecure": true,   <-- change to true`

**Reply by AlexeySa** [reply]
With such an option, the connection will be subject to MITM-attacks. I would like to use one certificate and use it to encrypt a connection with any SNI domain.

**Reply by kingwilliam** [reply]
if you use self-signed certificate, why not use wildcard certificate?
